File: src/statesman/commands/GotoCommand.java
package statesman.commands;

import statesman.Interpreter;

public class GotoCommand implements Command {

    public static final String Id = "goto";

    private String _groupName;

    public GotoCommand() {
        _groupName = "";
    }

    public GotoCommand(String commandGroupName) {
        this();
        if (commandGroupName.isBlank()) {
            throw new IllegalArgumentException();
        }
        _groupName = commandGroupName;
    }

    @Override
    public Command createInstance(String[] arguments) {
        if (arguments.length == 2) {
            return new GotoCommand(arguments[1]);
        }
        return null;
    }

    @Override
    public void execute() {
        CommandGroup group = Interpreter.getSource().getCommandGroups().get(_groupName);
        // Local scene groups override the global group
        if (Interpreter.getCurrentScene() != null) {
            CommandGroup localGroup = Interpreter.getCurrentScene().getCommandGroups().get(_groupName);
            if (localGroup != null) {
                group = localGroup;
            }
        }
        
        if (group != null) {
            group.execute();
        }
    }

}

File: src/Commands/GotoCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Statesman.Commands
{
    public class GotoCommand : Command
    {
        public static readonly string ID = "goto";

        private string _groupName;

        public GotoCommand()
        {
            _groupName = "";
        }

        public GotoCommand(string commandGroupName)
                : this()
        {
            if (string.IsNullOrWhiteSpace(commandGroupName))
            {
                throw new ArgumentException();
            }
            _groupName = commandGroupName;
        }

        public override Command createInstance(string[] arguments)
        {
            if (arguments.Length == 2)
            {
                return new GotoCommand(arguments[1]);
            }
            return null;
        }

        public override void execute()
        {
            CommandGroup group = null;
            if (Content.getScript().getCommandGroups().ContainsKey(_groupName))
            {
                group = Content.getScript().getCommandGroups()[_groupName];
            }

            // Local scene groups override the global group
            if (Interpreter.getScene() != null)
            {
                CommandGroup localGroup = null;
                if (Interpreter.getScene().getCommandGroups().ContainsKey(_groupName))
                {
                    localGroup = Interpreter.getScene().getCommandGroups()[_groupName];
                }
                if (localGroup != null)
                {
                    group = localGroup;
                }
            }

            if (group != null)
            {
                group.execute();
            }
        }
    }
}
